kotlin/internal/klib/klib.bzl

```diff
@@ -0,0 +1,78 @@
+load("//kotlin/internal:defs.bzl", _KtKlibInfo = "KtKlibInfo", _TOOLCHAIN_TYPE = "TOOLCHAIN_TYPE")
+load("//kotlin/internal/utils:utils.bzl", "utils")
+
+def _kt_klib_library(ctx):
+    module_name = utils.derive_module_name(ctx)
+    builder_args = utils.init_args(ctx, "kt_klib_library", module_name)
+
+    klib = ctx.actions.declare_file("{}.klib".format(ctx.label.name))
+    outputs = [klib]
+
+    toolchains = ctx.toolchains[_TOOLCHAIN_TYPE]
+    deps_klibs = []
+    for dep in ctx.attr.deps:
+        deps_klibs.append(dep[_KtKlibInfo].klibs)
+    libraries = depset(transitive = deps_klibs)
+    builder_args.add_all("--sources", ctx.files.srcs)
+    builder_inputs, _, input_manifests = ctx.resolve_command(tools = [toolchains.kotlinbuilder, toolchains.konan_home])
+
+    builder_args.add("--strict_kotlin_deps", "off")
+    builder_args.add("--reduced_classpath_mode", "off")
+    builder_args.add("--output_klib", klib.path)
+
+    deps_klibs = []
+    for dep in ctx.attr.deps:
+        deps_klibs.append(dep[_KtKlibInfo].klibs)
+    libraries = depset(transitive = deps_klibs)
+    builder_args.add_all("--klibs", libraries, omit_if_empty = False)
+
+    # This will be a directory we need to propagate to the compiler
+    konan_home = toolchains.konan_home[DefaultInfo].files.to_list()[0]
```

Comment:
Rule of thumb: if you unpack a depset with `to_list`, you are doing it wrong.

Let's step back and ask the question: does starlark need to know about konan home?

Or can we derive it in the action? I suspect we could just pass the files into a well known place and work from there.

Reply:
Removed this and used `directory` from bazel-skylib and access konan.home from the toolchain and avoid using this depset all together.